### PR TITLE
Oss 2.4 release notes

### PR DESCRIPTION
**Pull request by lwandzura**
Closes https://github.com/influxdata/docs-v2/issues/4255

added release notes

### CODE REVIEW

File: content/influxdb/v2.4/reference/release-notes/influxdb.md [hunk 1]
```diff
@@ -8,6 +8,46 @@ menu:
...
+
+- Log the log level at startup.
+- Fix panic from `CreateIterator` in Flux queries.
+- Fix error-caching in `bufio.Writer`.
+- Remove `MATCHER_BEHAVIOR` environment variable.
+
+### Maintenance 
+
+- Upgrade to [Go 1.18.4](https://go.dev/doc/go1.18).
+- Upgrade to [Flux 0.177.0](/flux/v0.x/release-notes/#v01770-2022-08-01).
```
Comment (samhld) [inline content/influxdb/v2.4/reference/release-notes/influxdb.md:49]:
```suggestion
- Upgrade to [Flux 0.179.0](flux/v0.x/release-notes/#v01790-2022-08-15).
```

File: content/influxdb/v2.4/reference/release-notes/influxdb.md [hunk 2]
```diff
@@ -8,6 +8,46 @@ menu:
...
+- Promote the following functions from experimental.http.requests into the [`http.requests package`](/flux/v0.x/stdlib/http/requests/): 
+  - [http.requests.do()](/flux/v0.x/stdlib/http/requests/do/) 
+  - [http.requests.get()](/flux/v0.x/stdlib/http/requests/get/)
+  - [http.requests.peek()](/flux/v0.x/stdlib/http/requests/peek/)
+  - [http.requests.post()](/flux/v0.x/stdlib/http/requests/post/)
+- Promote experimental.bitwise into the [`bitwise package`](/flux/v0.x/stdlib/bitwise/). 
+- Add new [experimental.catch()](/flux/v0.x/stdlib/experimental/catch/) and [testing.shouldError()](/flux/v0.x/stdlib/testing/shoulderror/) functions.
+- Support conditional expressions, constants, and literals in vectorized [map()](flux/v0.x/stdlib/universe/map/).
+- Optimize [holtWinters() function](/flux/v0.x/stdlib/universe/holtwinters/) and redundant sorts.
+- Deprecate experimental.http.get() and experimental.csv.from().
```
Comment (kelseiv) [inline content/influxdb/v2.4/reference/release-notes/influxdb.md:37]:
```suggestion
- Deprecate `experimental.http.get()` and `experimental.csv.from()`.
```

**Review by kelseiv — APPROVED** [review]

File: content/influxdb/v2.4/reference/release-notes/influxdb.md [hunk 3]
```diff
@@ -8,6 +8,46 @@ menu:
...
+## v2.4 [2022-08-18]
+
+### Features
+
+- Support [InfluxDB 1.x dbrp](/influxdb/v2.4/query-data/influxql/dbrp) names as bucket IDs in `replications` API.
+- Add the [InfluxQL shell (REPL)](/influxdb/v2.4/reference/cli/influx/v1/shell/).
+- Change to operator model so that `admin` user has instance-level permissions without `operator` token.
+
+#### Flux advancement highlights
+- Add multiple new join functions to the [`join package`](/flux/v0.x/stdlib/join/), including [join.full()](/flux/v0.x/stdlib/join/full/).
```
Comment (sanderson) [inline content/influxdb/v2.4/reference/release-notes/influxdb.md:20]:
```suggestion
- Add the [`join package`](/flux/v0.x/stdlib/join/) to support inner joins, right outer joins, left outer joins, and full outer joins.
```

File: content/influxdb/v2.4/reference/release-notes/influxdb.md [hunk 4]
```diff
@@ -8,6 +8,46 @@ menu:
...
+- Promote experimental.to() to [influxdb.wideTo()](/flux/v0.x/stdlib/influxdata/influxdb/wideto/).
+- Add [`initialZero`](/flux/v0.x/stdlib/universe/derivative/#initialzero) parameter to [derivative() function](/flux/v0.x/stdlib/universe/derivative/).
```
Comment (sanderson) [inline content/influxdb/v2.4/reference/release-notes/influxdb.md:22]:
```suggestion
- Add [`initialZero`](/flux/v0.x/stdlib/universe/derivative/#initialzero) parameter to [`derivative()`](/flux/v0.x/stdlib/universe/derivative/).
```

File: content/influxdb/v2.4/reference/release-notes/influxdb.md [hunk 5]
```diff
@@ -8,6 +8,46 @@ menu:
...
+- Add builtin function `time` to the [`date package`](flux/v0.x/stdlib/date/) to convert `timeable` into `datetime`.
```
Comment (sanderson) [inline content/influxdb/v2.4/reference/release-notes/influxdb.md:23]:
```suggestion
- Add `time()` function to the [`date package`](flux/v0.x/stdlib/date/) to convert timeable types (time and duration) to time types.
```

File: content/influxdb/v2.4/reference/release-notes/influxdb.md [hunk 6]
```diff
@@ -8,6 +8,46 @@ menu:
...
+- Promote the following functions from experimental.array into the [`array package`](/flux/v0.x/stdlib/array/): 
+  - [array.concat()](/flux/v0.x/stdlib/array/concat/)
+  - [array.filter()](/flux/v0.x/stdlib/array/filter/)
+  - [array.map()](/flux/v0.x/stdlib/array/map/)
+- Promote the following functions from experimental.http.requests into the [`http.requests package`](/flux/v0.x/stdlib/http/requests/): 
+  - [http.requests.do()](/flux/v0.x/stdlib/http/requests/do/) 
+  - [http.requests.get()](/flux/v0.x/stdlib/http/requests/get/)
+  - [http.requests.peek()](/flux/v0.x/stdlib/http/requests/peek/)
+  - [http.requests.post()](/flux/v0.x/stdlib/http/requests/post/)
```
Comment (sanderson) [inline content/influxdb/v2.4/reference/release-notes/influxdb.md:32]:
```suggestion
  - [requests.do()](/flux/v0.x/stdlib/http/requests/do/) 
  - [requests.get()](/flux/v0.x/stdlib/http/requests/get/)
  - [requests.peek()](/flux/v0.x/stdlib/http/requests/peek/)
  - [requests.post()](/flux/v0.x/stdlib/http/requests/post/)
```

File: content/influxdb/v2.4/reference/release-notes/influxdb.md [hunk 7]
```diff
@@ -8,6 +8,46 @@ menu:
...
+- Promote experimental.bitwise into the [`bitwise package`](/flux/v0.x/stdlib/bitwise/).
```
Comment (sanderson) [inline content/influxdb/v2.4/reference/release-notes/influxdb.md:33]:
```suggestion
- Promote `experimental.bitwise()` into the [`bitwise` package](/flux/v0.x/stdlib/bitwise/). 
```

File: content/influxdb/v2.4/reference/release-notes/influxdb.md [hunk 8]
```diff
@@ -8,6 +8,46 @@ menu:
...
+  - [array.filter()](/flux/v0.x/stdlib/array/filter/)
+  - [array.map()](/flux/v0.x/stdlib/array/map/)
+- Promote the following functions from experimental.http.requests into the [`http.requests package`](/flux/v0.x/stdlib/http/requests/): 
+  - [http.requests.do()](/flux/v0.x/stdlib/http/requests/do/) 
+  - [http.requests.get()](/flux/v0.x/stdlib/http/requests/get/)
+  - [http.requests.peek()](/flux/v0.x/stdlib/http/requests/peek/)
+  - [http.requests.post()](/flux/v0.x/stdlib/http/requests/post/)
+- Promote experimental.bitwise into the [`bitwise package`](/flux/v0.x/stdlib/bitwise/). 
+- Add new [experimental.catch()](/flux/v0.x/stdlib/experimental/catch/) and [testing.shouldError()](/flux/v0.x/stdlib/testing/shoulderror/) functions.
+- Support conditional expressions, constants, and literals in vectorized [map()](flux/v0.x/stdlib/universe/map/).
```
Comment (sanderson) [inline content/influxdb/v2.4/reference/release-notes/influxdb.md:35]:
```suggestion
- Support conditional expressions, constants, and literals in vectorized [`map()`](flux/v0.x/stdlib/universe/map/).
```

File: content/influxdb/v2.4/reference/release-notes/influxdb.md [hunk 9]
```diff
@@ -8,6 +8,46 @@ menu:
...
+- Optimize [holtWinters() function](/flux/v0.x/stdlib/universe/holtwinters/) and redundant sorts.
```
Comment (sanderson) [inline content/influxdb/v2.4/reference/release-notes/influxdb.md:36]:
```suggestion
- Optimize [`holtWinters()`](/flux/v0.x/stdlib/universe/holtwinters/) and redundant sorts.
```

**Review by sanderson — CHANGES_REQUESTED** [review]

File: content/influxdb/v2.4/reference/release-notes/influxdb.md [hunk 10]
```diff
@@ -8,6 +8,48 @@ menu:
...
+## v2.4 [2022-08-18]
+
+### Features
+
+- Support [InfluxDB 1.x dbrp](/influxdb/v2.4/query-data/influxql/dbrp) names as bucket IDs in `replications` API.
+- Add the [InfluxQL shell (REPL)](/influxdb/v2.4/reference/cli/influx/v1/shell/).
+- Change to operator model so that `admin` user has instance-level permissions without `operator` token.
+
+#### Flux advancement highlights
+- Add the [`join package`](/flux/v0.x/stdlib/join/) to support inner joins, right outer joins, left outer joins, and full outer joins.
```
Comment (sanderson) [inline content/influxdb/v2.4/reference/release-notes/influxdb.md:20]:
```suggestion
- Add the [`join` package](/flux/v0.x/stdlib/join/) to support inner joins, right outer joins, left outer joins, and full outer joins.
```

File: content/influxdb/v2.4/reference/release-notes/influxdb.md [hunk 11]
```diff
@@ -8,6 +8,48 @@ menu:
...
+- Promote `experimental.to()` to [`influxdb.wideTo()`](/flux/v0.x/stdlib/influxdata/influxdb/wideto/).
+- Add [`initialZero`](/flux/v0.x/stdlib/universe/derivative/#initialzero) parameter to [`derivative()`](/flux/v0.x/stdlib/universe/derivative/).
+- Add `time()` function to the [`date package`](flux/v0.x/stdlib/date/) to convert timeable types (time and duration) to time types.
+- Promote the following functions from `experimental.array` into the [`array` package](/flux/v0.x/stdlib/array/): 
+  - [array.concat()](/flux/v0.x/stdlib/array/concat/)
+  - [array.filter()](/flux/v0.x/stdlib/array/filter/)
+  - [array.map()](/flux/v0.x/stdlib/array/map/)
```
Comment (sanderson) [inline content/influxdb/v2.4/reference/release-notes/influxdb.md:27]:
```suggestion
  - [`array.concat(`)](/flux/v0.x/stdlib/array/concat/)
  - [`array.filter()`](/flux/v0.x/stdlib/array/filter/)
  - [`array.map()`](/flux/v0.x/stdlib/array/map/)
```

File: content/influxdb/v2.4/reference/release-notes/influxdb.md [hunk 12]
```diff
@@ -8,6 +8,48 @@ menu:
...
+- Promote the following functions from the experimental `http.requests` package into the [`http/requests` package](/flux/v0.x/stdlib/http/requests/): 
+  - [requests.do()](/flux/v0.x/stdlib/http/requests/do/) 
+  - [requests.get()](/flux/v0.x/stdlib/http/requests/get/)
+  - [requests.peek()](/flux/v0.x/stdlib/http/requests/peek/)
+  - [requests.post()](/flux/v0.x/stdlib/http/requests/post/)
```
Comment (sanderson) [inline content/influxdb/v2.4/reference/release-notes/influxdb.md:32]:
```suggestion
  - [`requests.do()`](/flux/v0.x/stdlib/http/requests/do/) 
  - [`requests.get()`](/flux/v0.x/stdlib/http/requests/get/)
  - [`requests.peek()`](/flux/v0.x/stdlib/http/requests/peek/)
  - [`requests.post()`](/flux/v0.x/stdlib/http/requests/post/)
```

File: content/influxdb/v2.4/reference/release-notes/influxdb.md [hunk 13]
```diff
@@ -8,6 +8,48 @@ menu:
...
+- Promote `experimental.bitwise()` into the [`bitwise` package](/flux/v0.x/stdlib/bitwise/). 
+- Add new [experimental.catch()](/flux/v0.x/stdlib/experimental/catch/) and [testing.shouldError()](/flux/v0.x/stdlib/testing/shoulderror/) functions.
+- Support conditional expressions, constants, and literals in vectorized [`map()`](flux/v0.x/stdlib/universe/map/).
+- Optimize [`holtWinters()`](/flux/v0.x/stdlib/universe/holtwinters/) and redundant sorts.
+- Deprecate the following experimental functions:
+  - `http.get()`
+  - `csv.from()`.
```
Comment (sanderson) [inline content/influxdb/v2.4/reference/release-notes/influxdb.md:39]:
```suggestion
  - `csv.from()`
```

File: content/influxdb/v2.4/reference/release-notes/influxdb.md [hunk 14]
```diff
@@ -8,6 +8,48 @@ menu:
 weight: 101
 ---
 
+## v2.4 [2022-08-18]
+
+### Features
+
+- Support [InfluxDB 1.x dbrp](/influxdb/v2.4/query-data/influxql/dbrp) names as bucket IDs in `replications` API.
```
Comment (sanderson) [inline content/influxdb/v2.4/reference/release-notes/influxdb.md:15]:
```suggestion
- Support bucket names in the `replications` API and allow [InfluxDB 1.x DBRP](/influxdb/v2.4/query-data/influxql/dbrp) names as bucket names.
```

**Review by sanderson — CHANGES_REQUESTED** [review]
Sorry, a few more suggestions.

**Review by sanderson — APPROVED** [review]

File: content/influxdb/v2.4/reference/release-notes/influxdb.md [hunk 15]
```diff
@@ -8,6 +8,48 @@ menu:
...
+- Add the [InfluxQL shell (REPL)](/influxdb/v2.4/reference/cli/influx/v1/shell/).
+- Change to operator model so that `admin` user has instance-level permissions without `operator` token.
+
+#### Flux advancement highlights
+- Add the [`join` package](/flux/v0.x/stdlib/join/) to support inner joins, right outer joins, left outer joins, and full outer joins.
+- Promote `experimental.to()` to [`influxdb.wideTo()`](/flux/v0.x/stdlib/influxdata/influxdb/wideto/).
+- Add [`initialZero`](/flux/v0.x/stdlib/universe/derivative/#initialzero) parameter to [`derivative()`](/flux/v0.x/stdlib/universe/derivative/).
+- Add `time()` function to the [`date package`](flux/v0.x/stdlib/date/) to convert timeable types (time and duration) to time types.
+- Promote the following functions from `experimental.array` into the [`array` package](/flux/v0.x/stdlib/array/): 
+  - [`array.concat(`)](/flux/v0.x/stdlib/array/concat/)
```
Comment (sanderson) [inline content/influxdb/v2.4/reference/release-notes/influxdb.md:25]:
```suggestion
  - [`array.concat()`](/flux/v0.x/stdlib/array/concat/)
```

File: content/influxdb/v2.4/reference/release-notes/influxdb.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (lwandzura) [reply content/influxdb/v2.4/reference/release-notes/influxdb.md]: @sanderson is it timeable or time.able?

[the remark this answers is not in the document]
Comment (sanderson) [reply content/influxdb/v2.4/reference/release-notes/influxdb.md]: It's "timeable." It describes a type constraint in flux: https://docs.influxdata.com/flux/v0.x/spec/types/#timeable-constraint

These a types that can be cast into a time value.

File: content/influxdb/v2.4/reference/release-notes/influxdb.md [hunk 16]
```diff
@@ -8,6 +8,48 @@ menu:
...
+- Optimize [`holtWinters()`](/flux/v0.x/stdlib/universe/holtwinters/) and redundant sorts.
+- Deprecate the following experimental functions:
+  - `http.get()`
+  - `csv.from()`
+
+### Bug fixes
+
+- Log the log level at startup.
+- Fix panic from `CreateIterator` in Flux queries.
+- Fix error-caching in `bufio.Writer`.
```
Comment (r00t-) [inline content/influxdb/v2.4/reference/release-notes/influxdb.md:45]: this seems to have ended up here by mistake, no such change was merged for what i can see.
the original [draft](https://github.com/influxdata/docs-v2/issues/4255) references my MR https://github.com/influxdata/influxdb/pull/23127 which has not been merged.
the entry was re-worded to "Fix error-caching in `bufio.Writer`",
but bufio.Writer is part of the go standard library (in which has no related change is to be found.)

**Pull request merged**